src/theory/strings/extf_solver.cpp

```diff
@@ -735,7 +735,12 @@ Node ExtfSolver::getCurrentSubstitutionFor(int effort,
   if (effort >= 1 && n.getType().isStringLike())
   {
     Assert(effort < 3);
-    // normal forms
+    // Return self if the normal form has not been computed. This may happen
+    // for terms that are not relevant in the current context.
```

Comment:
would it make sense to assert that `n` is not relevant in the current context?

Reply:
Its not trivial to check this :)